### PR TITLE
Improved issue templates

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
## Motivation

GitHub is improving their support for advanced issue templates; these new ones appear and behave more like abbreviated forms, which makes it much easier for users and developers to communicate the minimal amount of information directed to their current needs. It also allows us to include tidbits of extra information (like nicer external links) to direct them in their search for answers.

You can view previews of the forms (minus the GitHub encoding header) by viewing the raw files through the web GUI: [bug report](https://github.com/catalystneuro/matnwb/blob/improved_issue_templates/.github/ISSUE_TEMPLATE/bug_report.yml), [feature request](https://github.com/catalystneuro/matnwb/blob/improved_issue_templates/.github/ISSUE_TEMPLATE/feature_request.yml), and [documentation](https://github.com/catalystneuro/matnwb/blob/improved_issue_templates/.github/ISSUE_TEMPLATE/documentation.yml)

Some examples particular to the NWBInspector and their outputs here: https://github.com/catalystneuro/nwbinspector/issues

@lawrence-mbf Now, particular to MatNWB there I have some questions for making as similar forms as possible to the Python based projects.

1) Is there a Contributing document separate from the Code of Conduct?
2) What are all the versions of MATLAB currently supported by MatNWB?
3) Python has a way of reporting all versions of all external packages (`pip freeze > output_file.txt`). Do you know of an analogous command in MATLAB? The alternative is to have the user manually report such things, but that can sometimes be a daunting, time-consuming task.
4) Are error traces still referred to as 'tracebacks' in MATLAB, or are they called something different?

### CODE REVIEW

**Comment by bendichter** [comment]
@CodyCBakerPhD could you please add screenshots of the forms?

**Comment by CodyCBakerPhD** [comment]
@bendichter The screenshots don't give the best view of interactivity - for that I'd recommend the links in the PR description or open an issue on the CatalystNeuro fork of NWBInspecot: https://github.com/catalystneuro/nwbinspector/issues, which are essentially the same (I standardize the majority of the appearance across MatNWB/HDMF/PyNWB/NWBCT/NWBInspector.

But here they are:

Bug report

image

Feature request

image

Documentation

image

**Comment by bendichter** [comment]
bug report should not mention pip

**Comment by bendichter** [comment]
otherwise lgtm

**Comment by CodyCBakerPhD** [comment]
> bug report should not mention pip

@bendichter Yeah, that's one of four things that this draft PR needs to address. Maybe you know some of these answers?

> Now, particular to MatNWB there I have some questions for making as similar forms as possible to the Python based projects.
>
>1) Is there a Contributing document separate from the Code of Conduct?
>2) What are all the versions of MATLAB currently supported by MatNWB? E.g., 2020b - 2021a, etc.
>3) Python has a way of reporting all versions of all external packages (`pip freeze > output_file.txt`). Do you know of an analogous command in MATLAB? The alternative is to have the user manually report such things, but that can sometimes be a daunting, time-consuming task.
>4) Are error traces still referred to as 'tracebacks' in MATLAB, or are they called something different?

**Comment by CodyCBakerPhD** [comment]
From meeting today

> Is there a Contributing document separate from the Code of Conduct?

No, which is fine.

> What are all the versions of MATLAB currently supported by MatNWB? E.g., 2020b - 2021a, etc.

Theoretically very far back. Just ask what version user has at time of error.

> Python has a way of reporting all versions of all external packages (pip freeze > output_file.txt). Do you know of an analogous command in MATLAB? The alternative is to have the user manually report such things, but that can sometimes be a daunting, time-consuming task.

N/A

> Are error traces still referred to as 'tracebacks' in MATLAB, or are they called something different?

Just refer to as as the 'entire error message'.

**Comment by CodyCBakerPhD** [comment]
@bendichter Requested changes made to the forms in reference to the discrepancies with Python environments.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        Feel free to include as little or as much as you think is relevant. This section will be automatically formatted into code, so no need for backticks.
+      render: matlab
+    validations:
+      required: true
+  - type: textarea
+    id: error_message
+    attributes:
+      label: Error Message
+      description: |
+        Please copy and paste the message produced by the error.
```
Comment (bendichter) [inline .github/ISSUE_TEMPLATE/bug_report.yml:35]:
```suggestion
        Please copy and paste the entire error message.
```

**Pull request merged**